### PR TITLE
Incoming runtime messages (ADR 0011)

### PR DESCRIPTION
**Pull request by kostko**
This implements incoming runtime messages as per ADR 0011.

TODO
* [x] Add queries.
* [x] Update executor commitments.
* [x] Update runtime host protocol.
* [x] Update Rust runtime support library.
* [x] Update commit processing pipeline.
* [x] Update consensus roothash service to clear processed messages from queue.
* [x] Tests.

### CODE REVIEW

**Comment by kostko** [comment]
Task linked: [CU-1zfmwm8 Incoming runtime messages](https://app.clickup.com/t/1zfmwm8)

File: go/consensus/tendermint/apps/roothash/roothash.go [hunk 1]
```diff
@@ -508,9 +516,6 @@ func (app *rootHashApplication) processRoundTimeout(ctx *tmapi.Context, state *r
 		return fmt.Errorf("failed to finalize block: %w", err)
 	}
 
-	if err = state.SetRuntimeState(ctx, rtState); err != nil {
```
Comment (pro-wh) [inline go/consensus/tendermint/apps/roothash/roothash.go:511]: :information_source: this is moved into the above tryFinalizeBlock. all two callsites of tryFinalizeBlock would do this

---

but this breaks some symmetry, where we obtain rtState at this level, but we do the 'cleanup' procedures one level down inside tryFinalizeBlock

**Review by pro-wh — COMMENTED** [review]
5/42

File: go/consensus/tendermint/apps/roothash/roothash.go [hunk 2]
```diff
@@ -560,6 +565,59 @@ func (app *rootHashApplication) tryFinalizeExecutorCommits(
...
+				return fmt.Errorf("failed to fetch incoming message queue: %w", err)
+			}
+			if inMsgsHash := message.InMessagesHash(msgs); !ec.Header.InMessagesHash.Equal(&inMsgsHash) {
+				ctx.Logger().Debug("finalized round contained invalid incoming message hash, failing instead",
+					"in_msgs_hash", inMsgsHash,
+					"ec_in_msgs_hash", *ec.Header.InMessagesHash,
+				)
+				// Make the round fail.
+				err = fmt.Errorf("finalized round contained invalid incoming message hash")
+				// TODO: All nodes contributing to this round should be penalized.
```
Comment (ptrus) [inline go/consensus/tendermint/apps/roothash/roothash.go:589]: Is there an issue opened for this?

File: go/registry/api/runtime.go [hunk 3]
```diff
@@ -250,6 +253,10 @@ type RuntimeStakingParameters struct {
 	// RewardSlashBadResultsRuntimePercent is the percentage of the reward obtained when slashing
 	// for incorrect results that is transferred to the runtime's account.
 	RewardSlashBadResultsRuntimePercent uint8 `json:"reward_bad_results,omitempty"`
+
+	// MinInMessageFee specifies the minimum fee that the incoming message must include for the
+	// message to be queued.
+	MinInMessageFee quantity.Quantity `json:"min_in_msg_fee,omitempty"`
```
Comment (ptrus) [inline go/registry/api/runtime.go:259]: Maybe `min_in_message_fee` to be consistent with `max_in_messages`

**Review by ptrus — APPROVED** [review]

File: go/consensus/tendermint/apps/roothash/roothash.go (hunk 2 above)
Comment (kostko) [reply go/consensus/tendermint/apps/roothash/roothash.go:589]: No, there is an internal task about adding slashing/suspension that I plan to work on next and I thought that should be part of it. Will open an issue as well.

**Pull request merged**